docker_daemon/check.py

```diff
@@ -472,17 +472,28 @@ def _extract_tag_value(self, entity, tag_name):
 
         return entity["_tag_values"][tag_name]
 
+    def _localhost_ecs_agent(self):
```

Comment:
This should be called only once, otherwise that 5-second timeout will be painful. One solution would be to store the result in self on top returning it (`self._is_ecs_agent_local` for example, you'll need to set it to None at init time first) and add: 
```python
if self._is_ecs_agent_local is not None:
    return self._is_ecs_agent_local
```
At the beginning of this method.